### PR TITLE
feat: Add integrity check

### PR DESCRIPTION
**Pull request by kalidux**
- Updated `actions/checkout` to v4.
- Added SHA256 checksum calculation and verification for built binaries to ensure file integrity.
- Included file architecture validation to ensure correct binaries are built for each target platform.
- Updated the release draft step to include links to SHA256 checksum files for each binary.

## Summary by CodeRabbit

- **New Features**
	- Enhanced GitHub Actions workflow for building and releasing the `beacond` binary.
	- Added SHA256 checksum generation for improved integrity verification of release artifacts.
	- Updated release notes to include SHA256 checksums for each binary.

- **Improvements**
	- Expanded build process with clearer comments and additional commands for better transparency.
	- Streamlined naming conventions for uploaded artifacts, ensuring consistency.

- **Usability Enhancements**
	- New column in release notes for direct links to SHA256 checksum files, simplifying user access.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -69,24 +69,32 @@ jobs:
           - binary: beacond
     runs-on: ${{ matrix.configs.runs-on }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v3
         with:
           go-version: "^1.22"
```
Comment (itsdevbear) [inline .github/workflows/release.yaml:75]:
```suggestion
          go-version: "^1.23"
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -155,11 +163,11 @@ jobs:
 
           ## Binaries
 
-          | System | Architecture | Binary | PGP Signature |
+          | System | Architecture | Binary | PGP Signature | SHA256 Checksum |
           |:---:|:---:|:---:|:---|
```
Comment (itsdevbear) [inline .github/workflows/release.yaml:167]:
```suggestion
          |:---:|:---:|:---:|:---:|:---|
```

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -69,24 +69,32 @@ jobs:
...
           make build-${{ matrix.configs.target-os }}-${{ matrix.configs.arch }}-${{ needs.extract-version.outputs.VERSION }}
+          # Display architecture of the build binary 
+          file build/bin/${{ matrix.build.binary }}-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.configs.target-os }}-${{ matrix.configs.arch }}
+          # Create tar.gz archive
           tar -czvf ${{ matrix.build.binary }}-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.configs.target-os }}-${{ matrix.configs.arch }}.tar.gz -C build/bin .
-      
-      # Upload binaries
+          # Calculate SHA256 checksum and save to .tar.gz.sha256 file
+          shasum -a 256 ${{ matrix.build.binary }}-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.configs.target-os }}-${{ matrix.configs.arch }}.tar.gz > ${{ matrix.build.binary }}-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.configs.target-os }}-${{ matrix.configs.arch }}.tar.gz.sha256
+
```
[the remark this answers is not in the document]
Comment (kalidux) [reply .github/workflows/release.yaml:90]: `shasum -a 256` is more generic and portable for cross-platform scripts targeting macOS and Linux, it is preinstalled on macOS and Linux. 
 `sha256sum` is not available by default on macOS

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -154,12 +162,11 @@ jobs:
...
 
           ## Binaries
-
-          | System | Architecture | Binary | PGP Signature |
-          |:---:|:---:|:---:|:---|
-          | <img src="https://simpleicons.org/icons/linux.svg" style="width: 32px;"/> | amd64 | [beacond-${{ env.VERSION }}-linux-amd64](https://github.com/${{ env.REPO_NAME }}/releases/download/${{ env.VERSION }}/beacond-${{ env.VERSION }}-linux-amd64.tar.gz) |
-          | <img src="https://simpleicons.org/icons/linux.svg" style="width: 32px;"/> | arm64 | [beacond-${{ env.VERSION }}-linux-arm64](https://github.com/${{ env.REPO_NAME }}/releases/download/${{ env.VERSION }}/beacond-${{ env.VERSION }}-linux-arm64.tar.gz) | 
-          | <img src="https://simpleicons.org/icons/apple.svg" style="width: 32px;"/> | arm64 | [beacond-${{ env.VERSION }}-darwin-arm64](https://github.com/${{ env.REPO_NAME }}/releases/download/${{ env.VERSION }}/beacond-${{ env.VERSION }}-darwin-arm64.tar.gz) |
+          | System | Architecture | Binary | PGP Signature | SHA256 Checksum |
+          |:---:|:---:|:---:|:---:|:---|
```
[the remark this answers is not in the document]
Comment (kalidux) [reply .github/workflows/release.yaml]: fixed

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -69,24 +69,32 @@ jobs:
           - binary: beacond
     runs-on: ${{ matrix.configs.runs-on }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v3
         with:
-          go-version: "^1.22"
+          go-version: "^1.23"
```
[the remark this answers is not in the document]
Comment (kalidux) [reply .github/workflows/release.yaml]: fixed

**Review by ocnc — APPROVED** [review]

**Pull request merged**